cfme/tests/test_rest.py

```diff
@@ -374,13 +374,485 @@ def test_vm_add_lifecycle_event(request, setup_a_provider, rest_api, vm, from_de
+    try:
+        template = rest_api.collections.templates[0]
+    except:
+        pytest.skip("There is no template to be refreshed")
+
+    old_refresh_dt = template.update_on
+    assert template.action.refresh()["success"], "Refresh was unsuccessful"
+    wait_for(
+        lambda: template.update_on != old_refresh_dt,
+        fail_func=template.reload,
+        num_sec=180,
+        delay=10,
+    )
+
+
+@pytest.fixture(scope="module", params=["resource_pools", "templates", "data_storages",
+    "clusters"])
+def rest_api_delete_service(request, rest_api):
+    if request.param == 'resource_pools':
+        return ("resource_pools", rest_api.collections.resource_pools)
+    elif request.param == 'templates':
+        return ("templates", rest_api.collections.templates)
+    elif request.param == 'data_storages':
+        return ("data_storages", rest_api.collections.data_storages)
+    elif request.param == 'clusters':
+        return ("clusters", rest_api.collections.clusters)
+
+
+def test_delete_service(rest_api_delete_service):
+    """Test deleting a service ["resource_pools", "templates", "data_storages", "clusters"]
+
+    Prerequisities:
+        * An appliance with ``/api`` available.
+
+    Steps:
+        * Retrieve list of entities using GET /api/<service> , pick the first one
+        * DELETE /api/<service>/<id> -> delete only one data store
+        * DELETE /api/<service> <- Insert a JSON with list of dicts containing ``href``s to
+            the services
+        * Repeat the DELETE query -> now it should return an ``ActiveRecord::RecordNotFound``.
+
+    Metadata:
+        test_flag: rest
+    """
+    service_name, api_service = rest_api_delete_service
+    assert "delete" in api_service
+
+    try:
+        delete_service = api_service[0]
+    except:
+        pytest.skip("There is no {} to be deleted".format(service_name))
+
+    delete_service.action.delete()
+    wait_for(
+        lambda: not api_service.find_by(id=delete_service.id),
+        num_sec=180,
+        delay=10,
+    )
+
+    services = [service.id for service in api_service]
+    api_service.action.delete(services)
+    with error.expected("ActiveRecord::RecordNotFound"):
+        api_service.action.delete(services)
+
+
 # TODO: Gradually remove and write separate tests for those when they get extended
 @pytest.mark.parametrize(
     "collection_name",
-    ["availability_zones", "clusters", "conditions", "data_stores", "events", "flavors", "groups",
-    "hosts", "policies", "policy_actions", "policy_profiles", "request_tasks", "requests",
-    "resource_pools", "roles", "security_groups", "servers", "service_requests", "tags", "tasks",
-    "templates", "users", "zones"])
+    ["availability_zones", "clusters", "conditions", "events", "flavors", "hosts", "policy_actions",
+    "request_tasks", "requests", "security_groups", "servers", "service_requests", "tags", "tasks",
+    "zones"])
 def test_query_simple_collections(rest_api, collection_name):
```

Comment:
The code below does something we discourage now, in that it skips a test which should never run, can you modify this at this stage as you're in here doing a lot of work, so that it uncollects instead of skips if the version is less than 5.4?